.vib/grafana-tempo/ginkgo/grafana_tempo_test.go

```diff
@@ -24,15 +25,27 @@ var _ = Describe("Grafana Tempo", func() {
 	})
 
 	Context("through its API", func() {
-		var queryFrontendIp string
+		var queryFrontendIp, serviceName string
+		var hasIP bool
+		var err error
 
 		BeforeEach(func() {
-			queryFrontendSvc, err := c.Services(*namespace).Get(ctx, "grafana-tempo-query-frontend", metav1.GetOptions{})
+			serviceName = "grafana-tempo-query-frontend"
+
+			hasIP, err = retry("isServiceReady", 6, 20*time.Second, func() (bool, error) {
+				return isServiceReady(ctx, c, serviceName)
+			})
+			if err != nil {
+				panic(fmt.Sprintf("There was an error checking whether the testing service had an IP assigned: %q", err))
+			}
+			Expect(hasIP).To(BeTrue())
+
+			queryFrontendSvc, err := c.Services(*namespace).Get(ctx, serviceName, metav1.GetOptions{})
 			if err != nil {
 				panic(fmt.Sprintf("There was an error retrieving the Query Frontend service: %q", err))
 			}
 
-			queryFrontendIp = queryFrontendSvc.Status.LoadBalancer.Ingress[0].IP
+			queryFrontendIp = returnValidHost(queryFrontendSvc.Status.LoadBalancer.Ingress[0])
```

Comment:
Hmm although this is rather personal preference, I think that we should avoid using `queryFrontendIp` for a variable that actually holds a Hostname now.

Feel free to update the name.